Removing :en in favor of default locale for duration#inspect

dmasur opened the pull request:
Hi there,

i have an app without english as available locale. So i got an error when we try to inspect something like 1.day. This is done automatically when we use the dalli cache.

I would like to change the :en to ::I18n.default_locale to be sure that this is always constant and is an available locale.

Tests are all green with this change.

Calculating 
:locale => :en                              2.024k i/100ms
:locale => ::I18n.default_locale   2.236k i/100ms

:locale => :en                                  25.758k (±26.3%) i/s -    117.392k
:locale => ::I18n.default_locale       26.311k (±18.1%) i/s -    127.452k 

Is there any other intention why this should be :en?

joermungandr commented:
+1

senny commented:
@dmasur can you add a test-case to prevent regressions?

dmasur commented:
@senny Thanks for your feedback. 

I added a test in the branch, but messed up the branches...

Here is the Test: https://github.com/dmasur/rails/commit/f43e7b874a3de74984054819c7fb82b6c0ff05c2

Should i make a new pull request for this?

senny commented on activesupport/lib/active_support/duration.rb:
```diff
@@ -91,7 +91,7 @@ def inspect #:nodoc:
         sort_by {|unit,  _ | [:years, :months, :days, :minutes, :seconds].index(unit)}.
         map     {|unit, val| "#{val} #{val == 1 ? unit.to_s.chop : unit.to_s}"}.
-        to_sentence(:locale => :en)
+        to_sentence(:locale => ::I18n.default_locale)
```
can you use 1.9 Hash syntax?

senny commented on activesupport/test/core_ext/duration_test.rb:
```diff
@@ -70,6 +70,14 @@ def test_inspect
+    I18n.default_locale = :de
+    I18n.backend.store_translations(:de, { support: { array: { last_word_connector: ' und ' } } })
+    assert_equal '10 years, 1 month und 1 day', (10.years + 1.month  + 1.day).inspect
+    I18n.default_locale = current_locale
```
this should go into an `ensure` block.

senny commented:
@dmasur can you force push a squashed version of this PR to your branch? We try to keep the changes of a patch in a single commit.

dmasur committed:
Removing :en in favor of I18n.default_locale in duration#inspect

Hi there,

i have an app without english as available locale. So i got an error when we try to inspect something like 1.day. This is done automatically when we use the dalli cache.

I would like to change the :en to ::I18n.default_locale to be sure that this is always constant and is an available locale.

Tests are all green with this change.

Calculating -------------------------------------
:locale => :en                              2.024k i/100ms
:locale => ::I18n.default_locale   2.236k i/100ms
-------------------------------------------------
:locale => :en                                  25.758k (±26.3%) i/s -    117.392k
:locale => ::I18n.default_locale       26.311k (±18.1%) i/s -    127.452k

dmasur commented:
@senny sure i pushed it as one commit.

senny commented:
@dmasur looking good. Thank you :yellow_heart:

robertomiranda commented:
@dmasur @senny nice :+1:
